### PR TITLE
Create a parent CMake project for building app bundles on CI

### PR DESCRIPTION
**Pull request by akoeplinger**
This allows us to not run the CMake configure step separately for each libraries test suite which speeds up the build.

Helps with https://github.com/dotnet/runtime/issues/58549

### CODE REVIEW

File: src/libraries/tests.proj [hunk 1]
```diff
@@ -332,6 +332,52 @@
...
+
+    <MakeDir Directories="$(TestArchiveTestsDir)" />
+
+    <XcodeBuildApp
+        TargetOS="$(TargetOS)"
+        Arch="$(TargetArchitecture)"
+        XcodeProjectPath="%(XcodeProjects.Identity)"
+        DevTeamProvisioning="$(DevTeamProvisioning)"
+        Optimized="True"
+        DestinationFolder="$(TestArchiveTestsDir)" /> <!-- TODO: TestArchiveTestsRoot is different for when IsFunctionalTest==true but this is only set in the context of the .csproj so we need to find another solution here -->
```
Comment (akoeplinger) [inline src/libraries/tests.proj:375]: @steveisok we'll need to fix this once we turn on functional tests for iOS again.

Comment (akoeplinger) [reply src/libraries/tests.proj:375]: hmm actually we're still running functional tests on simulators right? I guess it'll fail there :)

Comment (steveisok) [reply src/libraries/tests.proj:375]: Yes, we're only skipping functional tests on devices for now since mlaunch can't handle the return code we expect.

**Review by marek-safar — APPROVED** [review]

**Pull request merged**